src/loom-provider.ts

```diff
@@ -400,7 +428,36 @@ export class LoomProvider {
     }
   }
 
-  protected _simulateEmptyBlock(block: any = {}) {
+  private async _commitTransaction(
```

Comment:
Hmm, you shouldn't duplicate middleware functionality here... I'll extend the Client API to let you override the middleware for each tx, that way you can just create a set of Nonce/SignedTx middleware for each private key.

Reply:
So when PR #40 is merged you can do this instead:

```typescript
this._client.commitTxAsync<Transaction>(tx, { middleware: createDefaultTxMiddleware(this._client, privateKey) })
```